File: src/main/java/ru/serdtsev/homemoney/ReferencesResource.java
package ru.serdtsev.homemoney;

import org.springframework.transaction.annotation.Transactional;
import org.springframework.web.bind.annotation.PathVariable;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;
import ru.serdtsev.homemoney.dao.ReferencesDao;
import ru.serdtsev.homemoney.common.HmResponse;

import java.util.UUID;

@RestController
@RequestMapping("/api/{bsId}/references")
public class ReferencesResource {
  @RequestMapping("currencies")
  @Transactional
  public final HmResponse getCurrencies(@PathVariable UUID bsId) {
    return HmResponse.getOk(ReferencesDao.getCurrencies(bsId));
  }
}

File: src/main/java/ru/serdtsev/homemoney/ReferencesResource.kt
package ru.serdtsev.homemoney

import org.springframework.web.bind.annotation.PathVariable
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController
import ru.serdtsev.homemoney.dao.ReferencesDao
import ru.serdtsev.homemoney.dto.HmResponse
import java.util.*

@RestController
@RequestMapping("/api/{bsId}/references")
class ReferencesResource {
  @RequestMapping("currencies")
  fun getCurrencies(@PathVariable("bsId") bsId: UUID): HmResponse = HmResponse.getOk(ReferencesDao.getCurrencies(bsId))
}